spec/models/manageiq/providers/embedded_ansible/automation_manager/configuration_script_source_spec.rb

```diff
@@ -192,6 +163,113 @@ def files_in_repository(git_repo_dir)
     end
   end
 
+  describe "playbooks_in_git_repository" do
```

Comment:
This doesn't actually test that method anywhere, so this is a bit confusing as a describe name.

Reply:
So I see what you are saying "I guess", but does technically call the method as part of `build_record` (that calls `.create_in_provider`, which calls `.sync`).  At the same time I was also mostly focusing the output of the tests on that method, so that is why I didn't say I was testing `.sync` either.

Unfortunately, based on all of the dependencies just to test the method, I need to:

- create a git directory
- make sure it has files
- have a saved record so there is an ID
- sync so the `git_repository` record is created

Now, I could have done steps `1-3` and just called `#playbooks_in_git_repository` instead of sync, but I already had `build_record` which did this for me (along with some needed stubs), so that was part of why it is a bit confusing.

And since we were going to the trouble of doing a `sync` anyway, it was nice to confirm the resulting records in the DB anyway as a result.

* * *

I can change the name, add a comment with an explaination, or do something different to test, but I figured this was the easiest way to test the "intent of the method" without having to have a bunch of boiler plate to grok.